src/libraries/System.Net.Http/src/System/Net/Http/Headers/HttpHeaders.cs

```diff
@@ -1135,10 +1146,11 @@ internal static int GetStoreValuesIntoStringArray(HeaderDescriptor descriptor, o
                 return 1;
             }
 
-            int length = GetValueCount(info);
-
-            if (length > 0)
+            lock (info)
             {
+                int length = GetValueCount(info);
+                Debug.Assert(length > 0);
```

Comment:
Just curious, why `if (length > 0)` is removed -- it was possible before, but not possible now?

Reply:
This could have never been 0. The `if` should have been an assert the whole time.
The method right above is [already has an assert](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Net.Http/src/System/Net/Http/Headers/HttpHeaders.cs#L1113) instead of a branch.